lib/active_shipping/carriers/canada_post_pws.rb

```diff
@@ -175,36 +176,35 @@ def parse_services_response(response)
     end
 
     def parse_service_options_response(response)
-      doc = REXML::Document.new(response)
-      service_node = doc.elements['service']
-      service_code = service_node.get_text("service-code").to_s
-      service_name = service_node.get_text("service-name").to_s
-      options_node = service_node.elements['options']
-      unless options_node.blank?
-        option_nodes = options_node.elements.collect('option') { |node| node }
-        options = option_nodes.inject([]) do |result, node|
-          option = {
-            :code => node.get_text("option-code").to_s,
-            :name => node.get_text("option-name").to_s,
-            :required => node.get_text("mandatory").to_s == "false" ? false : true,
-            :qualifier_required => node.get_text("qualifier-required").to_s == "false" ? false : true
-          }
-          option[:qualifier_max] = node.get_text("qualifier-max").to_s.to_i if node.get_text("qualifier-max")
-          result << option
-          result
-        end
+      doc = Nokogiri.XML(response)
+      doc.remove_namespaces!
+
+      service_code = doc.root.at("service-code").text
+      service_name = doc.root.at("service-name").text
+
+      option_nodes = doc.root.xpath('options/option')
+      options = option_nodes.map do |node|
+        option = {
+          :code => node.at("option-code").text,
+          :name => node.at("option-name").text,
+          :required => node.at("mandatory").text == "false" ? false : true,
+          :qualifier_required => node.at("qualifier-required").text == "false" ? false : true
```

Comment:
You don't need the ternary here, `node.at("qualifier-required").text != "false"`. But I guess it's a bit harder to read. Or at least is less explicit.